vud.module

```diff
@@ -9,161 +9,36 @@
 module_load_include('inc', 'vud', 'vud.theme');
 
 /**
- * Implementation of hook_menu().
+ * @param $account
+ *    User Entity Object
+ * @param $entity \Drupal\Core\Entity\EntityInterface
+ *    Requested Entity Object
  */
-function vud_menu() {
-  $items = array();
-
-  $items['admin/config/search/voteupdown/advanced'] = array(
-    'title'  => 'General',
-    'type'   => MENU_DEFAULT_LOCAL_TASK,
-    'weight' => -10,
-  );
-
-  $items['vote/%/%/%/%/%/%'] = array(
-    'title'            => 'Vote',
-    'page callback'    => 'vud_vote',
-    'page arguments'   => array(1, 2, 3, 4, 5, 6),
-    'access callback'  => 'vud_access_callback',
-    'access arguments' => array('use vote up/down', 1, 2, 3, 4),
-    'type'             => MENU_CALLBACK,
-    'file'             => 'vud.theme.inc',
-  );
-
-  $items['votereset/%/%/%/%/%'] = array(
-    'title'            => 'Reset vote',
-    'page callback'    => 'vud_reset',
-    'page arguments'   => array(1, 2, 3, 4, 5),
-    'access callback'  => 'vud_access_callback',
-    'access arguments' => array('reset vote up/down votes', 1, 2, 3, 4),
-    'type'             => MENU_CALLBACK,
-    'file'             => 'vud.theme.inc',
-  );
-
-  $items['vud/%ctools_js/denied/%'] = array(
-    'title'            => 'Vote denied',
-    'page callback'    => 'vud_denied_vote',
-    'page arguments'   => array(1, 3),
-    'access arguments' => array('access content'),
-    'type'             => MENU_CALLBACK,
-  );
+function vud_can_vote($account, $entity){
+  return $account->hasPermission("add or remove points votes on {$entity->getEntityTypeId()}")
```

Comment:
As mentioned in one of the first comments here, this permission is not declared.